### PR TITLE
Chore: Replace `get_storage_var_address` with starknet_rs_core

### LINKED ISSUE #221: Replace get_storage_var_address
Replace `get_storage_var_address` function implementation with starknet_rs.
https://github.com/0xSpaceShard/starknet-devnet-rs/blob/bb22e929e6e6e666b9be4a5ce549933e08fd2ce7/crates/starknet/src/utils.rs#L18

### PR DESCRIPTION
**Pull request by RajeshRk18**
Resolves #221 

## Checklist:

- [x] Checked the relevant parts of [development docs](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#development)
- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the latest commit of the target branch (or merged it into my branch)
- [ ] Updated the docs if needed, including the [TODO section](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#todo-to-reach-feature-parity-with-the-pythonic-devnet)
- [x] Linked the [issues](https://github.com/0xSpaceShard/starknet-devnet-rs/issues) which this PR resolves
- [x] Updated the tests if needed; all passing ([execution info](https://github.com/0xSpaceShard/starknet-devnet-rs/?tab=readme-ov-file#test-execution))

### CODE REVIEW

**Comment by RajeshRk18** [comment]
> If not yet reviewable (and sure looks like it's not), please convert this PR to draft (by clicking on `Convert to draft`)

Sure! Still wip

**Comment by RajeshRk18** [comment]
@FabijanC 
Do we need new error type or we just unwrap whenever `get_storage_var_address`? I have seen whenever `DevNetResult` is returned, they are unerapped. The one I implemented doesn't work.

File: crates/starknet-devnet-core/src/utils.rs [hunk 1]
```diff
@@ -1,8 +1,12 @@
 use blockifier::versioned_constants::VersionedConstants;
+use cairo_felt::Felt252;
 use serde_json::Value;
+use sha3::{Digest, Keccak256, Keccak256Core};
```
[the remark this answers is not in the document]
Comment (RajeshRk18) [reply crates/starknet-devnet-core/src/utils.rs:4]: In the linked [file](https://github.com/0xSpaceShard/starknet-devnet-rs/blob/bb22e929e6e6e666b9be4a5ce549933e08fd2ce7/crates%2Fstarknet%2Fsrc%2Futils.rs#L2), keccak has been used from lamdaworks starknet crate. So, I ported the impl here

[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet-devnet-core/src/utils.rs:4]: I don't understand; what is lamdaworks starknet crate?

[the remark this answers is not in the document]
Comment (RajeshRk18) [reply crates/starknet-devnet-core/src/utils.rs:4]: You can see starknet_in_rust dep in the linked file

[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet-devnet-core/src/utils.rs:4]: I think we no longer use starknet_in_rust, is your link to an old revision of the code?

[the remark this answers is not in the document]
Comment (RajeshRk18) [reply crates/starknet-devnet-core/src/utils.rs:4]: Yes, issue of this pr referenced that. `get_storage_var_address` needed keccak. So, there is keccak fn in src/utils.rs

[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet-devnet-core/src/utils.rs:4]: Doesn't [the issue](https://github.com/0xSpaceShard/starknet-devnet-rs/issues/221) require using the `get_storage_var_address` function from starknet_rs? And not adding a new custom implementation?

[the remark this answers is not in the document]
Comment (RajeshRk18) [reply crates/starknet-devnet-core/src/utils.rs:4]: "Replace with implementation", I thought just port it. Ok, I'll change

[the remark this answers is not in the document]
Comment (FabijanC) [reply crates/starknet-devnet-core/src/utils.rs:4]: Ok maybe my understanding was wrong. @marioiordanov, as the issue author, what is supposed to be done here?

[the remark this answers is not in the document]
Comment (RajeshRk18) [reply crates/starknet-devnet-core/src/utils.rs:4]: Let me know what needs to be done. Need to finish it off within today

[the remark this answers is not in the document]
Comment (marioiordanov) [reply crates/starknet-devnet-core/src/utils.rs:4]: Have to replace the get_storage_var_address function with starknet-rs-core::utils::get_storage_var_address

[the remark this answers is not in the document]
Comment (RajeshRk18) [reply crates/starknet-devnet-core/src/utils.rs:4]: Did you see what I've done? Is that right

**Comment by FabijanC** [comment]
Closing. Explanation in https://github.com/0xSpaceShard/starknet-devnet-rs/issues/221#issuecomment-2084598114

**Pull request closed without merging**